contribute/design/1.x/csi/20190606-csi-volume-provisioning.md

```diff
@@ -254,134 +247,141 @@ Below represents resources that are in a constant state of reconciliation
 ```
                     |--- CStorVolume
                     |   
-CStorVolumeClaim ---|--- ConfigInjector
-                    |
-                    |--- CStorVolumeReplica
-
-
-                    |--- CStor Target
+CStorVolumeConfig --|
                     |
-                    |--- CStor Service
-ConfigInjector -----|
                     |--- CStorVolumeReplica
-                    |
-                    |--- CStorVolume
 ```
 
 Next sections provide details on these resources that work in tandem to ensure 
 smooth volume create & delete operations.
 
-#### CStorVolumeConfigClass -- new custom resource
-This is a new Kubernetes custom resource that holds config information to be
-used by cstor volume claim controller while provisioning cstor volumes.
+#### CStorVolumePolicy -- new custom resource
+This resource kind does not have a controller i.e. watcher logic. It holds
+the config information to be utilized by cstor volume config controller.
 
-NOTE: This resource kind does not have a controller i.e. watcher logic. It holds
-the config information to be utilized by cstor volume claim controller.
+Following is the proposed schema for `CStorVolumePolicy`:
+```
+type CStorVolumePolicySpec struct {
+        // TargetSpec represents configuration related to cstor target and its resources
+        Target TargetSpec `json:"target"`
+        // ReplicaSpec represents configuration related to replicas resources
+        Replica ReplicaSpec `json:"replica"`
+}
 
-Following is the proposed schema for `CStorVolumeConfigClass`:
+// TargetSpec represents configuration related to cstor target and its resources
+type TargetSpec struct {
+        
+        // QueueDepth sets the queue size at iSCSI target which limits the
+        // ongoing IO count from client
+        QueueDepth string `json:"queueDepth,omitempty"`
+        
+        // Luworkers sets the number of threads that are working on above queue
+        LuWorkers int64 `json:"luWorkers,omitempty"`
+        
+        // Monitor enables or disables the target exporter sidecar
+        Monitor bool `json:"monitor,omitempty"`
```

Comment:
Why this field required? I think we shouldn't remove the containers based on the users input.

Reply:
Even if we might not allow to edit this field but it is required at the time of provisioning.

Reply:
Is there any case where users not required of a volume exporter sidecar? What is the reason behind giving a configurable option?